javascript/javascript2/week1/part1/main.js

```diff
@@ -0,0 +1,86 @@
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+
+function getShortestWord(wordsArr) {
+  if (wordsArr.length == 0) {
+    return "Words not found!";
+  }
+  let shortWord = wordsArr[0];
+  let shortWordLength = wordsArr[0].length;
+  for (let i = 0; i < wordsArr.length; i++) {
+    if (wordsArr[i].length < shortWordLength) {
+      shortWord = wordsArr[i];
+    }
+  }
+  return shortWord;
+}
+console.log(getShortestWord(danishWords)); // returns 'ø'
+
+const danishLetters = ["å", "æ", "ø"];
+
+function findDanishLetters(danishString) {
+  let result = { total: 0 };
+  for (let i = 0; i < danishString.length; i++) {
+    if (danishLetters.includes(danishString[i].toLowerCase())) {
+      result.total++;
+      if (result.hasOwnProperty(danishString[i].toLowerCase())) {
+        result[danishString[i].toLowerCase()] =
+          result[danishString[i].toLowerCase()] + 1;
+      } else {
+        result[danishString[i].toLowerCase()] = 1;
+      }
+    }
+  }
+  console.log(result);
+}
+
+const danishString = "Jeg har en blå bil";
+findDanishLetters(danishString); // returns {total: 1, å: 1}
+
+const danishString2 = "Blå grØd med røde bær";
+findDanishLetters(danishString2); // returns {total: 4, æ: 1, ø: 2, å: 1}
+
+const spiritAnimalsList = [
+  "Butterfly",
+  "Owl",
+  "Spider",
+  "Crow",
+  "Hummingbird",
+  "Hawk",
+  "Grasshopper",
+  "Eagle",
+  "Bat",
+  "Dolphin",
+];
+
+const generateButton = document.querySelector("button");
+const nameInput = document.querySelector("input");
+const generateSelect = document.getElementById("generate-settings");
+
+const namePTag = document.querySelector(".spirit-animal-name");
+
+function printSpiritName() {
+  const randomSpiritName =
+    spiritAnimalsList[Math.floor(Math.random() * spiritAnimalsList.length)];
+  if (nameInput.value.length > 0) {
+    namePTag.innerText = nameInput.value + " - " + randomSpiritName;
+  } else {
+    namePTag.innerText = "Enter name please!";
+  }
+}
+generateButton.onclick = function (event) {
```

Comment:
Cool very well done, works great ! 

Although, in the class you were going through addEventListener and this was probably an exercise to practice it. 
And just something to be aware of "onclick" is a property of the object(your button) , and can handle only single event for the element. So if you wanted to add more than one "action" to the button the original one would be overwritten. 

Hugs, "Lucia Eagle" 🐔

Reply:
@Falafelqueen here I'm a little confused. generateButton.onclick = function (event) {} this one will override orginal "action" and  generateButton.addEventListener("click", myFunction); and this one not? So betetr use EventListener?

Reply:
I can well see why it would be confusing because in this case it doesn't make a difference. But you can try it out yourself, to see the difference better :

```
// now only thing that happens is that the button turns red when you click it, it stops printing out the spirit animal 

generateButton.onclick = function () {
  if (generateSelect.value == "click") {
    printSpiritName();
  }
};

generateButton.onclick = function () {
  if (generateSelect.value == "click") {
    generateButton.style.background = "red";
  }
};
```
```
// here the button turns red on click and prints the spirit animal at the same time

generateButton.addEventListener("click", function(){
 if (generateSelect.value == "click") {
   printSpiritName();
 }
})


generateButton.addEventListener("click", function () {
  if (generateSelect.value == "click") {
    generateButton.style.background = "red";
  }
});
```

This might be good to aware of in case you are having a longer code, so you don't try to accidentally assign to events with .onclick to the same element and cancel out the previous.

Reply:
@Falafelqueen wow. Thank you very much. I got it 🙏  😊